exercises/08-exercise.py

```diff
@@ -31,26 +33,62 @@ def __init__(self, resource_path, chunk_size=1024):
             user specifices size of chunk for data, default is 1024
         '''
         self.chunks_of_data = []
-        for chunk in pd.read_csv(resource_path, chunksize=chunk_size, 
-                names =['time (s)', 'time (ns)', 'index', 'counts'], 
-                delimiter = " ", header=None):
-            chunk['adc'] = chunk['counts'].apply(adc2counts)
+        chunk = [data for data in pd.read_csv(resource_path, 
+            chunksize=chunk_size, delimiter = " ", header = None) ]
+        
+        num_cols = len(chunk[0].columns)
+
+        columns = ['time (s)', 'time (ns)', 'index']
+        columns_leftover = num_cols - len(columns)
+        columns = columns + [f'adc {i}' for i in range(columns_leftover)]
+        
+        for chunk in chunk:
+            chunk.columns = columns
+            
+            for column in range(columns_leftover):
+                chunk.iloc[:, column + 3] = chunk.iloc[:, column + 3].apply(adc2counts)
+            
             chunk['timestamp'] = chunk['time (s)'] + 1e-9*chunk['time (ns)']
-            chunk = chunk.drop(columns = ['time (s)', 'time (ns)', 'index', 'counts'])
-            chunk = chunk[['timestamp','adc']]
-            self.chunks_of_data.append(chunk)
-
-    def __call__(self, chunk_num):
+            chunk = chunk.drop(columns = ['time (s)', 'time (ns)', 'index'])
+            new_cols = chunk.columns.tolist()
+            new_cols = new_cols[-1:] + new_cols[:-1]
+            chunk = chunk[new_cols]
+            self.chunks_of_data.append(chunk)    
+        
+
+    def __call__(self, chunk_num, column):
         '''
         Returns 
         -------
         result: dataframe object
             specified chunk number/index from list of all chunks created
         '''
-        result = self.chunks_of_data[chunk_num]
-        return result
-
-class PizzaBoxEnHandler(HandlerBase):
+        
+        if column == 0:
+            cols = {'timestamp': self.chunks_of_data[chunk_num]['timestamp'],\
```

Comment:
why are you checking for `column==0`?
`column=0` should be the first adc, and column=n the nth adc

Reply:
```py
# suppose we have " timestamp, adc 0, adc 1, adc 2, adc 3 " and the user wants adc column 2 (adc 2), but the  
# length of the column names is 5, indexed from 0 to 4, so in order to get 'adc 2', we would need 2 + 1, which is 
# index 3, which returns the correct column, that's how I was thinking about it

column_desired = self.chunks_of_data[chunk_num].columns[column+1]
```